interface/views.py

```diff
@@ -71,7 +71,12 @@ def upload(request, course_code, assignment_code):
         form = UploadFileForm(request.POST, request.FILES)
         if form.is_valid():
             file = request.FILES['file']
-            handle_submission(file, assignment, request.user)
+            delta_t = handle_submission(file, assignment, request.user)
+
+            if delta_t:
```

Comment:
Let's model this case as an exception so that it's clear this is an error situation. Define an exception in `submission.py`, e.g.

```python
class TooManySubmissionsError(Exception):
    pass
```

Throw it from `handle_submission` and catch it here.